flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/common/InfluxParser.java

```diff
@@ -134,7 +132,8 @@ private Number parseTimestamp(@Nullable final TimestampContext timestamp) {
         if (strTimestamp.length() < 7) {
             return 0L;
         } else {
-            return Long.valueOf(strTimestamp.substring(0, strTimestamp.length() - 6));
+            //            return Long.valueOf(strTimestamp.substring(0, strTimestamp.length() - 6));
+            return Long.valueOf(strTimestamp);
```

Comment:
@1p4pk We should talk about timestamp parsing. I don't see the point to do the substring.

Reply:
I thought we had removed that already. Or was it only in the benchmark branch?

Reply:
Yes, we only removed that in the benchmark branch. Just checked

Reply:
Then remove the line before
```suggestion
            return Long.valueOf(strTimestamp);
```